Fix building without Perfetto (series 0001): run_vm_tests only needs
libprotozero when Perfetto is supported.
Upstream: landed in dart-lang/sdk as e4dec8c8967 (#64525), after 3.13.
--- runtime/bin/BUILD.gn.orig	2026-09-29 08:00:52 UTC
+++ runtime/bin/BUILD.gn
@@ -1122,10 +1122,10 @@ source_set("run_vm_tests_set") {
     configs += [ "..:dart_precompiler_config" ]
   }
 
-  deps = [
-    "//third_party/boringssl",  # for secure_socket_utils_test
-    "//third_party/perfetto:libprotozero",  # for timeline_test
-  ]
+  deps = [ "//third_party/boringssl" ]  # for secure_socket_utils_test
+  if (dart_support_perfetto) {
+    deps += [ "//third_party/perfetto:libprotozero" ]  # for timeline_test
+  }
 
   if (is_fuchsia) {
     deps += [
@@ -1187,9 +1187,11 @@ executable("run_vm_tests") {
     ":run_vm_tests_set",
     "..:libdart_precompiler_testing",
     "//third_party/boringssl",  # for secure_socket_utils_test
-    "//third_party/perfetto:libprotozero",  # for timeline_test
     "//third_party/zlib",
   ]
+  if (dart_support_perfetto) {
+    deps += [ "//third_party/perfetto:libprotozero" ]  # for timeline_test
+  }
 
   if (is_win) {
     libs = [
